Applications/Tizen_native/CustomShortcut/src/view.c

```diff
@@ -224,15 +227,62 @@ static void _on_draw_end(void *data, Evas *e, Evas_Object *obj,
   evas_object_image_data_update_add(ad->canvas, 0, 0, ad->width, ad->height);
 }
 
-static Eina_Bool _on_canvas_exit(void *data, Elm_Object_Item *it) {
-  _D("exiting canvas");
+static void _on_canvas_exit(void *data, Evas *e, Evas_Object *obj,
+                            void *event_info) {
+  _D("deleting canvas");
   appdata_s *ad = (appdata_s *)data;
-  /// @todo save file to loadable data format
 
   evas_object_del(ad->canvas);
-  cairo_surface_destroy(ad->cr_surface);
   cairo_destroy(ad->cr);
-  return EINA_TRUE;
+  if (cairo_status(ad->cr) != CAIRO_STATUS_SUCCESS) {
+    _E("delete cairo failed");
+  }
+  cairo_surface_destroy(ad->cr_surface);
+  if (cairo_surface_status(ad->cr_surface) != CAIRO_STATUS_SUCCESS) {
+    _E("delete cr_surface failed");
+  }
+}
+
+static void _canvas_erase_all(appdata_s *ad) {
+  cairo_set_source_rgba(ad->cr, 0.5, 0.5, 0.5, 0.5);
+  cairo_set_operator(ad->cr, CAIRO_OPERATOR_SOURCE);
+  cairo_paint(ad->cr);
+  cairo_surface_flush(ad->cr_surface);
+  evas_object_image_data_update_add(ad->canvas, 0, 0, ad->width, ad->height);
+}
+
+static void _on_draw_reset(void *data, Evas_Object *obj, const char *emission,
+                           const char *source) {
+  appdata_s *ad = (appdata_s *)data;
+  _D("draw reset");
+  _canvas_erase_all(ad);
+}
+
+static void _on_draw_proceed(void *data, Evas_Object *obj, const char *emission,
+                             const char *source) {
+  appdata_s *ad = (appdata_s *)data;
+  int status = APP_ERROR_NONE;
+  _D("draw proceed");
```

Comment:
how about using different name _D, _E, _I. I don't think it is good name of function.

Reply:
Changed :)